### PR TITLE
Add Option to Login From Donate View & Refactor Login View Into Partials

### PR DESCRIPTION
**Pull request by adam2451**

### CODE REVIEW

File: features/donations/quick_donation.feature [hunk 1]
```diff
@@ -9,11 +9,11 @@ Background:
   Given the following account codes exist:
   | name             | code | description                                                     | donation_prompt          |
   | Soda Fund        | 0504 | The Soda Funds aims to put a Soda Fountain in Soda Hall         |                          |
-  And I am logged in as customer "Tom Foolery"
```
Comment (PratyushSharma14) [inline features/donations/quick_donation.feature:12]: Removed Background Step to automatically login as a User as Web Step that checks if User is not logged in doesn't work as intended,
Therefore, each test in first line will specify if there's a User logged in, or no login

File: features/donations/quick_donation.feature [hunk 2]
```diff
@@ -96,27 +96,45 @@ Scenario: admin logged in, records donation on behalf of patron
...
   When I go to the quick donation page
   Then I should not see "Donate to"
   And I should see "General Fund"
   And I should see "General Fund Address"
   
 Scenario: landing on quick donation page with valid account code and making quick donation
+  Given I am logged in as customer "Tom Foolery"
   When I go to the quick donation page
-  When I fill in "Donation amount" with "15"
+  And I fill in "Donation amount" with "15"
```
Comment (PratyushSharma14) [inline features/donations/quick_donation.feature:122]: Steps cannot have multiple "When" and "Then" statements togethor, should use "And" instead

File: app/views/layouts/_login_form.haml
Comment (MayZamudio) [inline app/views/layouts/_login_form.haml:1]: - Capitalize: Login for a Faster Checkout! So it matches the title case in Login page
- add a little bit of padding on the row to match the style throughout the Donation view and the New Customer card in login view
image
image
image

**Review by MayZamudio — COMMENTED** [review]
Looks really good just have a couple of bootstrap style recommendations :))

File: features/donations/quick_donation.feature [hunk 3]
```diff
@@ -96,27 +96,45 @@ Scenario: admin logged in, records donation on behalf of patron
...
 Scenario: landing on quick donation page with invalid account code
+  Given I am logged in as customer "Tom Foolery"
   When I visit the quick donation landing page for account code 0505
   Then I should see "Invalid Fund ID"
 
 Scenario: landing on quick donation page with no account code
+  Given I am logged in as customer "Tom Foolery"
   When I go to the quick donation page
   Then I should not see "Donate to"
   And I should see "General Fund"
```
Comment (armandofox) [inline features/donations/quick_donation.feature:116]: I point out that lines 116 and 117 are redundant.  if line 117 is true, line 116 will of necessity be true.  this is a big peril of using "Then I should see": it is true if the content occurs ANYWHERE on the page.  is there a way to scope this a bit more tightly, and if not, at least eliminate the redundant line?

**Review by armandofox — APPROVED** [review]

**Pull request merged**